### PR TITLE
Update JavaScript tests to use TypeScript

### PR DESCRIPTION
**Pull request by pepone**
This PR updates JavaScript test suite to use TypeScript. It also simplifies the generation of TypeScript definitions (.d.ts) files.

### CODE REVIEW

File: cpp/src/slice2js/Gen.cpp [hunk 1]
```diff
@@ -2268,234 +2215,231 @@ Slice::Gen::TypeScriptImportVisitor::visitDictionary(const DictionaryPtr& dict)
...
-    ClassDefPtr base = p->base();
-    if (base)
+    ClassDeclPtr cl = dynamic_pointer_cast<ClassDecl>(type);
+    if (cl)
     {
-        addAlias(dynamic_pointer_cast<Type>(base->declaration()), module);
+        t = importPrefix(cl->scoped()) + fixId(cl->scoped());
+        if (nullable)
+        {
+            t = _iceImportPrefix + "Ice.Nullable<" + t + ">";
```
Comment (externl) [inline cpp/src/slice2js/Gen.cpp:2326]: We have our own nullable? We should probably get rid of this we can,

Comment (pepone) [reply cpp/src/slice2js/Gen.cpp:2326]: Is an alias for `T | null`

Comment (externl) [reply cpp/src/slice2js/Gen.cpp:2326]: Why have it though?

Comment (pepone) [reply cpp/src/slice2js/Gen.cpp:2326]: Not strong feelings here, we can go either way. Is just a minor detail.

**Review by externl — APPROVED** [review]

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: cpp/src/slice2js/Gen.cpp (hunk 1 above)
Comment (pepone) [reply cpp/src/slice2js/Gen.cpp:2326]: I will fix it in a follow-up PR.

**Pull request merged**